api/proto/teleport/legacy/types/types.proto

```diff
@@ -6861,6 +6864,18 @@ message AWSICUserSyncFilter {
   map<string, string> labels = 8 [(gogoproto.jsontag) = "labels,omitempty"];
 }
 
+// AWSSyncFilter represents a filter used to include or exclude AWS resource during sync based on the filter criteria.
+message AWSSyncFilter {
+  option (gogoproto.equal) = true;
+  // Include specifies resources to include during synchronization.
+  oneof include {
+    // Filter by ID.
+    string id = 1;
+    // Filter by Name using a regex.
+    string name = 2;
```

Comment:
Will we support multiple name regexp filter or this will always be one field? Asking because this gets called in the repeated field.

Reply:
Also when both ID and name are configured, which field takes precedence? Can we write a comment here how the filtering will be handled?

Reply:
This is one of filed so setting both `id` and `name` by design  are not possible.

Reply:
Thinking of a direct translation to json/yaml it could make sense to do something like:

```
string field = 1;
string value = 2;
```

instead